kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/metadata/node/config/processor/impl/AbstractChangedJobConfigurationProcessor.java

```diff
@@ -38,16 +38,16 @@ public abstract class AbstractChangedJobConfigurationProcessor implements Change
     @Override
     public void process(final Type eventType, final JobConfiguration jobConfig) {
         boolean disabled = jobConfig.isDisabled();
-        if (disabled) {
-            onDisabled(jobConfig);
-        }
         boolean deleted = Type.DELETED == eventType;
         if (deleted) {
             onDeleted(jobConfig);
         }
         String jobId = jobConfig.getJobName();
         if (disabled || deleted) {
             PipelineJobCenter.stop(jobId);
+            if (disabled) {
+                onDisabled(jobConfig);
+            }
```

Comment:
Could `onDisabled` be called before job stopping?

Reply:
if `onDisabled` be called before job stopping, may have the problem that when execute start job command, the DistSQL return the result but the job is block at stopping.

![image](https://user-images.githubusercontent.com/101622833/227721125-7b9f24bd-0e49-4949-8517-1698ffbd5d96.png)
https://github.com/apache/shardingsphere/actions/runs/4510078482/jobs/7941413185

Reply:
How it occur?
`onDisabled` just persist ephemeral node by `PipelineDistributedBarrier`, and `added to executing jobs failed since it already exists` code block is after `onDisabled` invocation, no matter in original code and modified code of this PR.

Reply:
Because this PR add the `org.apache.shardingsphere.data.pipeline.core.job.AbstractPipelineJob#awaitJobStopped`, it's maybe blocked unitil elastic job really stopped. 

when PipelineDistributedBarrier persist all ephemeral node, DistSQL result will return, but the job info still at PipelineJobCenter.JOB_MAP
```
    private void awaitJobStopped(final PipelineElasticJobListener jobListener, final String jobId, final long timeoutMillis) {
        int time = 0;
        int sleepTime = 50;
        while (time < timeoutMillis) {
            if (!jobListener.isJobRunning(jobId)) {
                break;
            }
            try {
                Thread.sleep(sleepTime);
            } catch (final InterruptedException ignored) {
                break;
            }
            time += sleepTime;
        }
    }
```

At first commit, `sleepTime` = 1000, if the user executes `start migration` DistSql within 1000 milliseconds immediately after the `stop migration` DistSQL returns,  the problem will occur, so I adjusted sleepTime to 50

Reply:
OK